src/UnicycleOptimization.cpp

```diff
@@ -8,23 +8,37 @@
 #include "UnicycleOptimization.h"
 #include "iDynTree/Core/VectorFixSize.h"
 
+void UnicycleOptimization::updateState(const iDynTree::Vector2 &rPl, double deltaAngle, double deltaTime, const iDynTree::Vector2 &newStepPosition, const iDynTree::Vector2 &newStepRelativePosition)
+{
+    m_stateBuffer(0) = rPl(0);
```

Comment:
A bit complex to debug, but ok :)

Reply:
``rPl`` is the position of the left foot expressed in the frame attached to the right frame.